backend/api/serializers/task_profiling.py

```diff
@@ -82,7 +83,11 @@ class Meta:
             "duration",
         ]
 
-    def get_duration(self, function: Function) -> str:
+    def get_duration(self, function: Function) -> Union[str, None]:
+        counter_steps = len(orchestrator.FunctionProfilingStep)
+
+        if function.profiling_steps.count() < counter_steps:
+            return None
```

Comment:
We wait until all steps are completed before returning something? 
I guess it probably warrants a comment, at least :)

Reply:
Yes, mimicking what is in place for the task profiling: https://github.com/Substra/substra-backend/blob/a9bf510d3916d18dc1f6baec2826f2742ec27cb8/backend/api/serializers/task_profiling.py#L41-L45

We show the task total duration only if the task duration is finished (the frontend use this information to display percents of the task only when the task is finished)